### DISCUSSION TITLE
Is there a list of supported websites Tandoor can import recipes from?

### DISCUSSION THREAD

**Discussion by DonDebonair** [TandoorRecipes/recipes#3243] [General]
I want to understand what websites Tandoor can import recipes from? Are there pre-configured parsing rules for supported recipe websites, or are imports done based on heuristics?

**Comment by smilerz** [comment]
Tandoor uses the [recipe-scrapers](https://github.com/hhursev/recipe-scrapers) library which generically will scrape any recipe in [ld+json](https://developers.google.com/search/docs/appearance/structured-data/recipe) format.  They also have custom scrapers for sites that do not follow or embellish upon the standard format.

**Reply by DonDebonair** [reply]
Thanks for the swift reply!

If a site doesn't use ld+json to include recipe data, do you parse HTML?

**Reply by hurricane-socrates** [reply]
cooked.wiki

**Reply by smilerz** [reply]
Tandoor exclusively uses the recipe-scrapers library.  

That library, if ld+json is not available, will fail unless someone creates a custom scraper for that specific site.

**Comment by delta0815** [comment]
Thanks for your explanation. I came to this post because I tried to import this recipe and ran into an error: https://www.rewe.de/rezepte/gemuese-erdnuss-wok/

As you can see it's from rewe.de and recipe-scrapers has it listed on their site. I am just wondering if Tandoor isn't using the most current library or if rewe.de changed their syntax and it's not Tandoor failing to scrape it, but the recipe-scrapers library that needs to be updated.

Thanks for your support

Cheers,

Oliver

**Reply by smilerz** [reply]
you can see what version of the library by reviewing the requirements.txt from the release of Tandoor you are using.

**Reply by delta0815** [reply]
Thanks a lot. I'm using the docker-image which seems to be using 15.4.0 and it seems that the scrapers library made a change to the rewe.de implementation in 15.5.0... so I guess I'll wait until Tandoor is updating the library and hope that it will start to work then :-)
Thanks a lot and cheers!
